src/main/java/com/google/api/codegen/go/GoGapicContext.java

```diff
@@ -345,64 +357,102 @@ private GoImport createMessageImport(MessageType messageType) {
-        imports.add(createMessageImport(outputMessage));
+        messageImports.add(createMessageImport(outputMessage));
       }
       if (methodConfig.isPageStreaming()) {
         TypeRef resourceType = methodConfig.getPageStreaming().getResourcesField().getType();
         if (resourceType.isMessage()) {
-          imports.add(createMessageImport(resourceType.getMessageType()));
+          messageImports.add(createMessageImport(resourceType.getMessageType()));
         }
-        coreImports.add(GoImport.create("io"));
       }
     }
 
-    List<String> result = new ArrayList<>();
-    for (GoImport goImport : coreImports) {
-      result.add("\t" + goImport.importString());
+    return messageImports;
+  }
+
+  private List<String> formatImports(Set<GoImport> standard, Set<GoImport> thirdParty) {
+    List<String> result = new ArrayList<String>();
+    if (standard != null) {
+      for (GoImport goImport : standard) {
+        result.add("    " + goImport.importString());
+      }
+      result.add("");
     }
-    // An empty string to bring the blank line between core imports and others.
-    result.add("");
-    for (GoImport goImport : imports) {
-      result.add("\t" + goImport.importString());
+    if (thirdParty != null) {
+      for (GoImport goImport : thirdParty) {
+        result.add("    " + goImport.importString());
+      }
     }
     return result;
   }
 
   /**
+   * Calculates the set of imports and returns a sorted set of Go import output strings. This
+   * imitates the same order which gofmt does, which means: - standard imports (Go standard libraries)
+   * in alphabetical order - a blank line (so an empty string) - other imports, alphabetical order
+   *
+   * Each of the lines (except for the blank line) starts with a tab character '\t' for the
+   * indentation within the 'import' section in Go file.
+   */
+  public Iterable<String> getImports(Interface service) {
+    TreeSet<GoImport> thirdParty = new TreeSet<>();
+
+    // Add non-service-specific imports.
+    thirdParty.add(GoImport.create("golang.org/x/net/context"));
+    thirdParty.add(GoImport.create("google.golang.org/grpc"));
+    thirdParty.add(GoImport.create("google.golang.org/grpc/codes"));
+    thirdParty.add(GoImport.create("google.golang.org/grpc/metadata"));
+    thirdParty.add(GoImport.create(GAX_PACKAGE_BASE, "gax"));
+
+    thirdParty.addAll(getMessageImports(service));
+    Set<GoImport> standard = getStandardImports(service);
+    return formatImports(standard, thirdParty);
+  }
+
+  /**
+   * Same as getImports, but scoped to the client_test.go file.
+   */
+  public Iterable<String> getTestImports(Interface service) {
+    TreeSet<GoImport> thirdParty = new TreeSet<>();
+
+    thirdParty.add(GoImport.create("golang.org/x/net/context"));
+    thirdParty.add(GoImport.create(GAX_PACKAGE_BASE, "gax"));
+    thirdParty.addAll(getMessageImports(service));
+
+    return formatImports(null, thirdParty);
+  }
```

Comment:
One motivation for the null is it prevents a newline space before the thirdParty imports.
I can modify the format function to ignore the newline if the standard imports set is empty, lmk if that's preferred.